### ISSUE TITLE
Node RED crashes when HA restarted

### ISSUE THREAD

**Issue by JasonBSteele** [zachowj/node-red-contrib-home-assistant-websocket#1243]
### Describe the bug

I have NR and HA running in separate containers on a Synology NAS.

When I restart HA so that updated integrations are used, NR also restarts. 

### To Reproduce

1. Restart HA
2. Check whether NR has restarted
3. Check the NR log for errors

### Expected behavior

I wouldn't expect NR to be restarted. I would hope that it would periodically retry to connect. 

### Environment Information

Version: 0.62.2

Home Assistant version: 2024.1.1
Companion version: 0.0.0 (v3.1.2, 0.0.0 was returned by the diagnostic button in NR)

Node-RED version: 3.1.3
Docker: yes
Add-on: no

Node.js version: v16.20.2 x64 linux
OS: Linux 4.4.302+ x64

### Additional context

The error in the NR log is:

```
2024/01/07 12:25:59,stdout,===================
2024/01/07 12:25:59,stdout,Welcome to Node-RED
2024/01/07 12:25:59,stdout,
2024/01/07 12:25:58,stdout,7 Jan 12:25:57 - [info] 
2024/01/07 12:25:28,stdout,"7 Jan 12:25:28 - [error] UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason \"3\".
"
2024/01/07 12:25:28,stdout,7 Jan 12:25:28 - [red] Uncaught Exception:
```

**Comment by aheath70** [comment]
Watching, as it is very similar to issue I logged recently. 

https://github.com/zachowj/node-red-contrib-home-assistant-websocket/issues/1242

**Comment by zachowj** [comment]
Can you add the code snip to the top of your NR `settings.js` file? So I can better understand exactly where this error is being thrown from?

```js
process.on('unhandledRejection', (reason, p) => {
  console.log('Unhandled Rejection: ', p, 'reason:', reason);
});
```

**Comment by aheath70** [comment]
Done. 

Log below. Note that the error no longer occurs and I can deploy successfully. I pulled the code back out again, restarted NodeRED and got the error again. 

```
Jan  9 21:48:51 nodered node-red[32232]: 9 Jan 21:48:51 - [info]
Jan  9 21:48:51 nodered node-red[32232]: Welcome to Node-RED
Jan  9 21:48:51 nodered node-red[32232]: ===================
Jan  9 21:48:51 nodered node-red[32232]: 9 Jan 21:48:51 - [info] Node-RED version: v3.1.3
Jan  9 21:48:51 nodered node-red[32232]: 9 Jan 21:48:51 - [info] Node.js  version: v18.17.1
Jan  9 21:48:51 nodered node-red[32232]: 9 Jan 21:48:51 - [info] Linux 6.5.11-7-pve x64 LE
Jan  9 21:48:52 nodered node-red[32232]: 9 Jan 21:48:52 - [info] Loading palette nodes
Jan  9 21:48:52 nodered node-red[32232]: 9 Jan 21:48:52 - [info] Settings file  : /root/.node-red/settings.js
Jan  9 21:48:52 nodered node-red[32232]: 9 Jan 21:48:52 - [info] Context store  : 'default' [module=memory]
Jan  9 21:48:52 nodered node-red[32232]: 9 Jan 21:48:52 - [info] User directory : /root/.node-red
Jan  9 21:48:52 nodered node-red[32232]: 9 Jan 21:48:52 - [warn] Projects disabled : editorTheme.projects.enabled=false
Jan  9 21:48:52 nodered node-red[32232]: 9 Jan 21:48:52 - [info] Flows file     : /root/.node-red/flows.json
Jan  9 21:48:53 nodered node-red[32232]: 9 Jan 21:48:53 - [info] Server now running at http://127.0.0.1:1880/
Jan  9 21:48:53 nodered node-red[32232]: 9 Jan 21:48:53 - [warn]
Jan  9 21:48:53 nodered node-red[32232]: ---------------------------------------------------------------------
Jan  9 21:48:53 nodered node-red[32232]: Your flow credentials file is encrypted using a system-generated key.
Jan  9 21:48:53 nodered node-red[32232]: If the system-generated key is lost for any reason, your credentials
Jan  9 21:48:53 nodered node-red[32232]: file will not be recoverable, you will have to delete it and re-enter
Jan  9 21:48:53 nodered node-red[32232]: your credentials.
Jan  9 21:48:53 nodered node-red[32232]: You should set your own key using the 'credentialSecret' option in
Jan  9 21:48:53 nodered node-red[32232]: your settings file. Node-RED will then re-encrypt your credentials
Jan  9 21:48:53 nodered node-red[32232]: file using your chosen key the next time you deploy a change.
Jan  9 21:48:53 nodered node-red[32232]: ---------------------------------------------------------------------
Jan  9 21:48:53 nodered node-red[32232]: 9 Jan 21:48:53 - [info] Starting flows
Jan  9 21:48:53 nodered node-red[32232]: 9 Jan 21:48:53 - [info] Started flows
Jan  9 21:48:53 nodered node-red[32232]: 9 Jan 21:48:53 - [info] [server:Home Assistant] Connecting to http://192.168.0.40:8123
Jan  9 21:48:53 nodered node-red[32232]: 9 Jan 21:48:53 - [info] [mqtt-broker:MQTT] Connected to broker: node-red.local@mqtt://mqtt.lan:1883
Jan  9 21:48:53 nodered node-red[32232]: 9 Jan 21:48:53 - [info] [server:Home Assistant] Connected to http://192.168.0.40:8123
Jan  9 21:49:29 nodered node-red[32232]: 9 Jan 21:49:29 - [info] Stopping flows
Jan  9 21:49:29 nodered node-red[32232]: 9 Jan 21:49:29 - [info] [server:Home Assistant] Closing connection to http://192.168.0.40:8123
Jan  9 21:49:29 nodered node-red[32232]: Unhandled Rejection:  Promise { <rejected> 3 } reason: 3
Jan  9 21:49:29 nodered node-red[32232]: 9 Jan 21:49:29 - [info] [mqtt-broker:MQTT] Disconnected from broker: node-red.local@mqtt://mqtt.lan:1883
Jan  9 21:49:29 nodered node-red[32232]: 9 Jan 21:49:29 - [info] Stopped flows
Jan  9 21:49:29 nodered node-red[32232]: 9 Jan 21:49:29 - [info] Updated flows
Jan  9 21:49:29 nodered node-red[32232]: 9 Jan 21:49:29 - [info] Starting flows
Jan  9 21:49:29 nodered node-red[32232]: 9 Jan 21:49:29 - [info] Started flows
Jan  9 21:49:29 nodered node-red[32232]: 9 Jan 21:49:29 - [info] [server:Home Assistant] Connecting to http://192.168.0.40:8123
Jan  9 21:49:30 nodered node-red[32232]: 9 Jan 21:49:30 - [info] [mqtt-broker:MQTT] Connected to broker: node-red.local@mqtt://mqtt.lan:1883
Jan  9 21:49:30 nodered node-red[32232]: 9 Jan 21:49:30 - [info] [server:Home Assistant] Connected to http://192.168.0.40:8123
Jan  9 21:49:30 nodered node-red[32232]: Unhandled Rejection:  Promise { <rejected> 3 } reason: 3
```

**Comment by JasonBSteele** [comment]
> Can you add the code snip to the top of your NR `settings.js` file? So I can better understand exactly where this error is being thrown from?
> 
> ```js
> process.on('unhandledRejection', (reason, p) => {
>   console.log('Unhandled Rejection: ', p, 'reason:', reason);
> });
> ```

Hi, should this be right at the top - or after `module.exports`?

**Comment by JasonBSteele** [comment]
Hi, I went with putting it before `module.exports`. It seemed to prevent me being able to view the logs in the Container Manager as  kept getting a timeout.

When I removed it and restarted I was able to view the logs again and could see that this time I didn't get the issue with NR crashing - I could see that it repeatedly tried Connecting to HA, and then when HA came back it reconnected fine.

When I removed the extra lines from settings.js and tried again I got the error as originally reported and NR crashed.

I tried putting the lines back again. This time I was able to view the log but it wasn't updating it had stopped just before the log entry that is usually given of `[info] Settings file : /data/settings.js`
